In [2]:
import numpy as np
import matplotlib.pyplot as plt
import math
from scipy import constants
import sympy as sy

ModuleNotFoundError: No module named 'sympy'

### Question D-2

We create the following two functions to complete the goals of the problem:

wavenumbers_kilojoules
- Inputs: Wavenumbers in cm $^{-1}$
- Ouputs: Kilojoules per Mol (KJ/mol)

wavenumbers_kilocalories
- Inputs: Wavenumbers in cm $^{-1}$
- Ouputs: Kilocalories per Mol (KJ/mol)

Our conversations that we use are found on the following website: http://wild.life.nctu.edu.tw/class/common/energy-unit-conv-table.html

In [ ]:
h2_vibrational_constant = 4401 # vibrational constant for H2 in cm^{-1}
h2_rotational_constant = 59.3 # rotational constant for H2 in cm^{-1}!

# functions which are described above in the associated markdown
def wavenumbers_kilojoules(wavenumbers): # Has to be in cm^-1
    return wavenumbers*0.0119627 # finds the conversion 

def wavenumbers_kilocalories(wavenumbers): # Has to be in cm^-1
    return wavenumbers*0.00285911	 # finds the conversion

print("H2 Vibrational Constant in Kilojoules: ", wavenumbers_kilojoules(h2_vibrational_constant), " kJ/mol")
print("H2 Rotational Constant in Kilojoules: ", wavenumbers_kilojoules(h2_rotational_constant), " kJ/mol")
print("H2 Vibrational Constant in Kilojoules: ", wavenumbers_kilocalories(h2_vibrational_constant), " kcal/mol")
print("H2 Rotational Constant in Kilojoules: ", wavenumbers_kilocalories(h2_rotational_constant), " kcal/mol")


H2 Vibrational Constant in Kilojoules:  52.6478427  kJ/mol
H2 Rotational Constant in Kilojoules:  0.7093881099999999  kJ/mol
H2 Vibrational Constant in Kilojoules:  12.58294311  kcal/mol
H2 Rotational Constant in Kilojoules:  0.169545223  kcal/mol


### Question D-4

Below, I calculate the numerical difference between $v=0,J=1$ and $v=1,J=0$ levels of this molecule using the experimental value. I take the code I used in HW2 to complete this task, therefore I have omitted the markdown sections that explain the code sections since I have already done that. 

In [ ]:
def transenergy(n,m,L): #translational energy, particle in a box; mass in kg, box length in m
  return (n**2 * constants.Planck**2)/(8*m*L**2)

def rotenergy(J,mu,bondlength): #rotational energy in terms of reduced mass (kg) and effective bond length (m)
  return J*(J+1)*constants.hbar**2/(2*mu*bondlength**2)

def vibenergy(v,mu,k): #vibrational energy in terms of force constant (N/m or kg sec^2) and reduced mass (kg)
  return (v+0.5)*constants.Planck*(1/(2*math.pi))*sy.sqrt(k/mu)

In [ ]:
# Below I calculate the rotational states of CO!

mass_carbon = 1.9944733e-26 # kg
mass_oxygen = 2.6566962e-26  # kg

molecule_reduced = (mass_carbon*mass_oxygen)/(mass_carbon+mass_oxygen)

J = np.array([0,1]) # the two rotational states that we care about
bondlength = 1.275e-10 #

rot_energies_molecule = rotenergy(J,molecule_reduced,bondlength)

print("Here are the two rotational energies of CO that we desire:")
i=0
for energy in rot_energies_molecule:
    print(i,") ", energy, "J")
    i += 1
i = 0

Here are the two rotational energies of CO that we desire:
0 )  0.0 J
1 )  6.005151906575091e-23 J


In [ ]:
# Below I calculate the rotational states of H^35Cl!
v = np.array([0,1])
# v_obs = 440100/ constants.Planck
k = ((2*math.pi*288600*constants.c)**2)*molecule_reduced
vib_energies_molecule = vibenergy(v,molecule_reduced,k)

print("Here are the first two vibrational energies of CO in Joules:")
i = 0
for energy in vib_energies_molecule:
    print(i,") ", energy, "J")
    i += 1

Here are the first five vibrational energies of H2 in Joules:
0 )  2.86644137186590e-20 J
1 )  8.59932411559771e-20 J


In [ ]:
# Below I calculate the Vibrational states of CO!
v = np.array([0,1])
# v_obs = 440100/ constants.Planck
k = ((2*math.pi*288600*constants.c)**2)*molecule_reduced
vib_energies_molecule = vibenergy(v,molecule_reduced,k)

print("Here are the first five vibrational energies of H2 in Joules:")
i = 0
for energy in vib_energies_molecule:
    print(i,") ", energy, "J")
    i += 1

Here are the first five vibrational energies of H2 in Joules:
0 )  2.86644137186590e-20 J
1 )  8.59932411559771e-20 J


In [ ]:
# Below, we compare the energies of v=0,J=1 and v=1,J=0

energy_total_1 = vib_energies_molecule[0] + rot_energies_molecule[1] # v=0,J=1
energy_total_2 = vib_energies_molecule[1] + rot_energies_molecule[0] # v=1,J=0

energy_difference = energy_total_1 - energy_total_2

print("The energy difference between E_(v=0,J=1)-E_(v=1,J=0)=", energy_difference)
print("\n Where the energy of E_(v=1,J=0)>E_(v=0,J=1)")

The energy difference between E_(v=0,J=1)-E_(v=1,J=0)= -5.72687759182523e-20

 Where the energy of E_(v=1,J=0)>E_(v=0,J=1)


The energy difference between $E_{v=0,J=1}-E_{v=1,J=0}= -5.73 \times 10^{-20}$ Joules, where the energy of $E_{v=1,J=0}>E_{v=0,J=1}$.

In [5]:
rotational_constant = constants.Planck**2 / (8 * constants.pi**2 * 6.68e-46 * constants.Boltzmann)
print(rotational_constant)

0.6029240133140447
